src/zql/integration.test.ts

```diff
@@ -0,0 +1,316 @@
+import {expect, test} from 'vitest';
+import {z} from 'zod';
+import {generate} from '../generate.js';
+import {makeReplicacheContext} from './context/replicache-context.js';
+import {Replicache, TEST_LICENSE_KEY} from 'replicache';
+import {nanoid} from 'nanoid';
+import {EntityQuery} from './query/entity-query.js';
+import fc from 'fast-check';
+
+const issueSchema = z.object({
+  id: z.string(),
+  title: z.string(),
+  status: z.enum(['open', 'closed']),
+  priority: z.enum(['high', 'medium', 'low']),
+  assignee: z.string(),
+  created: z.date(),
+  updated: z.date(),
+  closed: z.date().optional(),
+});
+
+type Issue = z.infer<typeof issueSchema>;
+
+const {
+  init: initIssue,
+  set: setIssue,
+  update: updateIssue,
+  delete: deleteIssue,
+} = generate<Issue>('issue', issueSchema.parse);
+
+const mutators = {
+  initIssue,
+  setIssue,
+  updateIssue,
+  deleteIssue,
+};
+
+function newRep() {
+  return new Replicache({
+    licenseKey: TEST_LICENSE_KEY,
+    name: nanoid(),
+    mutators,
+  });
+}
+
+const issueArbitrary: fc.Arbitrary<Issue> = fc.record({
+  id: fc.string({
+    minLength: 1,
+    maxLength: 10,
+  }),
+  title: fc.string(),
+  status: fc.constantFrom('open', 'closed'),
+  priority: fc.constantFrom('high', 'medium', 'low'),
+  assignee: fc.string(),
+  created: fc.date(),
+  updated: fc.date(),
+  closed: fc.option(fc.date(), {nil: undefined}),
+});
+
+// const operators = ['=', '<', '>', '>=', '<=', 'IN', 'LIKE', 'ILIKE'] as const;
+
+const tenUniqueIssues = fc.uniqueArray(issueArbitrary, {
+  comparator: (a, b) => a.id === b.id,
+  minLength: 10,
+  maxLength: 10,
+});
+
+function sampleTenUniqueIssues() {
+  return fc.sample(tenUniqueIssues, 1)[0];
+}
+
+function setup() {
+  const r = newRep();
+  const c = makeReplicacheContext(r);
+  const q = new EntityQuery<{fields: Issue}>(c, 'issue');
+  return {r, c, q};
+}
+
+// why is it backwards?
+const compareIds = (a: {id: string}, b: {id: string}) =>
+  a.id < b.id ? 1 : a.id > b.id ? -1 : 0;
+
+const compareIdsAsc = (a: {id: string}, b: {id: string}) =>
+  a.id < b.id ? -1 : a.id > b.id ? 1 : 0;
+
+// This test fails because `experimentalWatch` does not call us with an empty array when we want initial data from an empty collection.
```

Comment:
Fun bug.

Reply:
Seems to be working correctly. Something else is going on here.

Reply:
I found some tests for this case and they pass. I also added some tests here for it (spying on experimental watch) and those work too. Not sure what's going on yet.